### PR TITLE
[grafana] Upgrade PodDisruptionBudget apiVersion

### PR DESCRIPTION
**Pull request by besha100**
PodDisruptionBudget apiVersion policy/v1beta1 is deprecated in v1.21+. and will be unavailable in the next Kubernetes version. Nothing else needs to be changed in the yaml specs

### CODE REVIEW

File: charts/grafana/templates/poddisruptionbudget.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.podDisruptionBudget }}
```
Comment (zanhsieh) [inline charts/grafana/templates/poddisruptionbudget.yaml:1]: Could you move this to templates/_helpers.tpl and define like this?

```
{{/*
Return the appropriate apiVersion for ingress.
*/}}
{{- define "grafana.psp.apiVersion" -}}
  {{- if and (.Capabilities.APIVersions.Has "policy/v1") (semverCompare ">= 1.21-1" .Capabilities.KubeVersion.Version) -}}
    {{- print "policy/v1" -}}
  {{- else -}}
    {{- print "policy/v1beta1 -}}
  {{- end -}}
{{- end -}}
```
and use as:
kind: {{ template "grafana.psp.apiVersion" . }}

File: charts/grafana/templates/podsecuritypolicy.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.rbac.pspEnabled }}
-apiVersion: policy/v1beta1
+apiVersion: policy/v1
```
Comment (zanhsieh) [inline charts/grafana/templates/podsecuritypolicy.yaml:2]: See above.

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Comment by besha100** [comment]
Thank you! It's done. Can you review @zanhsieh

**Comment by zanhsieh** [comment]
@besha100 
I think you need to bump the chart version to latest, as the lint test failed.

File: charts/grafana/templates/_helpers.tpl [hunk 3]
```diff
@@ -141,6 +141,17 @@ Return the appropriate apiVersion for ingress.
...
 {{- end -}}
 
+{{/*
+Return the appropriate apiVersion for podsecuritypolicy and PodDisruptionBudget.
+*/}}
+{{- define "grafana.policy.apiVersion" -}}
+  {{- if and (.Capabilities.APIVersions.Has "policy/v1") (semverCompare ">= 1.21-1" .Capabilities.KubeVersion.Version) -}}
+    {{- print "policy/v1" -}}
+  {{- else -}}
+    {{- print "policy/v1beta1 -}}
```
Comment (zanhsieh) [inline charts/grafana/templates/_helpers.tpl:151]: Missing close double quote here. Should be:
```
{{- print "policy/v1beta1" -}}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/grafana/templates/_helpers.tpl (hunk 3 above)
Comment (besha100) [reply charts/grafana/templates/_helpers.tpl]: @zanhsieh Fixed! Thank you :)

**Comment by besha100** [comment]
@zanhsieh now suceeded. Thank you

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**